src/Dumplie/Domain/Customer/CartItem.php

```diff
@@ -36,6 +36,14 @@ public function __construct(Product $product, int $quantity)
     }
 
     /**
+     * @return Product
+     */
+    public function product() : Product
```

Comment:
I think we gonna need to replace Product with CartItem and remove price from it completely, I can't see any reason why to keep it in the basket.